pycloudlib/azure/instance.py

```diff
@@ -208,6 +211,17 @@ def start(self, wait=True):
             self.wait()
         self._status = VMInstanceStatus.ACTIVE
 
+    def _wait_for_instance_start(self, **kwargs):
```

Comment:
Why is this method needed now?

Reply:
1. The default one in the abstract class does nothing.
2.  (more important) From my testing I noticed that sometimes during `start` we try accessing/using the vm before it is up. This is due to the fact that the azure `start` method is async i.e. [`begin_start` ](https://github.com/canonical/pycloudlib/blob/fec76cc7e52b4fd5397c8528727eade5ac159935/pycloudlib/azure/instance.py#L202) hence we do not confirm that it is up before running the various cloudinit commands. With this we can confirm that the vm is up and running.

Reply:
Gotcha, thanks for adding this!